### ISSUE TITLE
对结果的一点疑问

### ISSUE THREAD

**Issue by wn1652400018** [Xuanfang1121/CRASpell_pytorch#2]
看您的结果如下：
precision:0.7484026201754532, recall:0.8443791867735061, f1_score:0.7934992640496943
然后又写到绝对acc为0.7372.
我理解这里的f1_score是指句子中所有错误的词都得到正确纠正才算正确，然后计算的句子级的F指。也就是之前大多是论文里采用的评估方式。
绝对的acc是什么有点不太清楚。是指总正确纠正的字/所有错别字吗？
不知道上面两个理解对不对。

f1_score:0.7934992640496943是不是和图片最后一列是一种评估方式。（因为CRASpell里给出指标都是按照字符来的，这里希望弄清楚一点，希望作者不要嫌我啰嗦。）
![图片](https://user-images.githubusercontent.com/35511759/179747664-6da5686a-e87a-408a-986d-a194c528184d.png)

**Comment by wn1652400018** [comment]
看了下评估的代码。发现您给出的指标的计算方式是按照字符级的，也就是和论文里一致，不过和论文里的指标相差比较多。
请问这是您精心调参后的结果吗？

**Comment by Xuanfang1121** [comment]
论文中作者使用的关于错别字的预训练模型，没有公开，我使用的是chinese-roberta-wwm-ext，所以结果上会差很多

**Comment by Xuanfang1121** [comment]
> 看了下评估的代码。发现您给出的指标的计算方式是按照字符级的，也就是和论文里一致，不过和论文里的指标相差比较多。 请问这是您精心调参后的结果吗？

论文中作者使用的关于错别字的预训练模型，没有公开，我使用的是chinese-roberta-wwm-ext，所以结果上会差很多

**Comment by Xuanfang1121** [comment]
> 看了下评估的代码。发现您给出的指标的计算方式是按照字符级的，也就是和论文里一致，不过和论文里的指标相差比较多。 请问这是您精心调参后的结果吗？

绝对acc 是指预测的句子和真实的句子完全一致记为正确，统计正确的个数/总的真实句子的数量

**Comment by wn1652400018** [comment]
> 
作者在‘2. Instructions’的第一行给了一个链接，应该就是作者用的cBert吧。
https://drive.google.com/file/d/1cqSTpn7r9pnDcvMoM3BbX1X67JsPdZ8_/view?usp=sharing

**Comment by wn1652400018** [comment]
我目前跑了一个epoch，结果如下：
`Epoch 1 step 8500,  train loss = 0.0205,0.0197,0.0112 ( 0.0982 examples/sec; 1.5712 sec/batch)
multi-error result:
zi result:
/home/wangning/workspace/CRASpell/datas/trained_models/CRACSpell/pred.txt
token num: gold_n:273, pred_n:247, right_n:215
token check: p=0.870, r=0.788, f=0.827
token correction-1: p=0.930, r=0.733, f=0.820
token correction-2: p=0.810, r=0.733, f=0.769
overall result:
zi result:
/home/wangning/workspace/CRASpell/datas/trained_models/CRACSpell/pred.txt
token num: gold_n:682, pred_n:775, right_n:598
token check: p=0.772, r=0.877, f=0.821
token correction-1: p=0.958, r=0.840, f=0.895
token correction-2: p=0.739, r=0.840, f=0.787`

我想请教一下，是不是correction-2，0.787，对应的是作者论文提出的评估方式。
而correction-1的precision评估感觉不太对（把原本正确但是却改错的忽略了，导致该值虚高）。

**Comment by Xuanfang1121** [comment]
> > 
> 
> 作者在‘2. Instructions’的第一行给了一个链接，应该就是作者用的cBert吧。 https://drive.google.com/file/d/1cqSTpn7r9pnDcvMoM3BbX1X67JsPdZ8_/view?usp=sharing

这个tensorflow的，没有相对应的torch版本的cBert

**Comment by yycberton** [comment]
你好，你训练的模型文件可以分享一下吗？

**Comment by Xuanfang1121** [comment]
> 你好，你训练的模型文件可以分享一下吗？
不好意思，模型文件被删了
